measurements/geography_based.py

```diff
@@ -97,6 +160,66 @@ def geo_tag(dataloader, args):
     info_stats['tag_to_subregion_features'] = tag_to_subregion_features
     pickle.dump(info_stats, open("results/{}/geo_tag.pkl".format(args.folder), "wb"))
 
+# private function called from geo_tag() if dataset is of gps form
+def geo_tag_gps(dataloader, args):
+    # map from a region name to a list whose value at index i represents count of category
+    # i 
+    region_tags = {}
+    tag_to_region_features = {}
+    categories = dataloader.dataset.categories
+
+    if not os.path.exists("results/{}/geo_ctr_gps.pkl".format(args.folder)):
+        print('running geo_ctr_gps() first to get necessary info...')
+        geo_ctr_gps(dataloader, args)
+    
+    counts_gps = pickle.load(open("results/{}/geo_ctr_gps.pkl".format(args.folder), "rb"))
+    id_to_region = counts_gps['id_to_region']
+
+    # get name of regions
+    unique_regions = list(set(id_to_region.values()))
+
+    # Extracts features from model pretrained on ImageNet
+    use_cuda = torch.cuda.is_available()
+    device = torch.device("cuda" if use_cuda else "cpu")
+    model = models.alexnet(pretrained=True).to(device)
+    new_classifier = nn.Sequential(*list(model.classifier.children())[:-1])
+    model.classifier = new_classifier
+    normalize = transforms.Normalize(mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225])
+
+    region_features = {}
+    for region in unique_regions:
+        region_features[region] = []
+
+    for cat in range(len(categories)):
+        tag_to_region_features[cat] = copy.deepcopy(region_features)
+
+    for i, (data, target) in enumerate(tqdm(dataloader)):
+        if data is None:
```

Comment:
when is data None?

Reply:
I don't think it ever is but just copied the functionality from [geo_tag](https://github.com/princetonvisualai/revise-tool/blob/441223d33f9047af77630a3dd0be8f1bd87dc2f0/measurements/geography_based.py#L74) and [geo_lng](https://github.com/princetonvisualai/revise-tool/blob/441223d33f9047af77630a3dd0be8f1bd87dc2f0/measurements/geography_based.py#L100). let me know if I should remove though